synapseutils/sync.py

```diff
@@ -975,3 +975,49 @@ def _check_size_each_file(df):
             single_file_size = os.stat(os.path.expandvars(os.path.expanduser(file_path))).st_size
             if single_file_size == 0:
                 raise ValueError("File {} is empty, empty files cannot be uploaded to Synapse".format(file_name))
+
+
+def generate_sync_manifest(syn, directory_path, parent_id, manifest_path):
+    """Generate manifest for syncToSynapse() from a local directory."""
+    manifest_cols = ["path", "parent"]
+    manifest_rows = _walk_directory_tree(syn, directory_path, parent_id)
+    _write_manifest_data(manifest_path, manifest_cols, manifest_rows)
+
+
+def _create_folder(syn, name, parent_id):
+    """Create Synapse folder."""
+    entity = {
+        'name': name,
+        'concreteType': 'org.sagebionetworks.repo.model.Folder',
+        'parentId': parent_id
+    }
+    entity = syn.store(entity)
```

Comment:
Currently if the folder already exists, it will simply create an "update" of the existing folder and return the existing Synapse id.

Reply:
I wasn't sure what to expect if I re-ran this function with the same inputs. Indeed, it updates the folder, evidenced by the changing `etag` and `modifiedOn` values. Practically, the folder doesn't change on Synapse.org and there doesn't seem to be any artificial versions being created, evidenced by the last command below. 

This behavior seems fine to me. Do you agree, @thomasyu888?

```
>>> import synapseclient
>>> syn = synapseclient.login()
>>> from synapseutils.sync import _create_folder

>>> out = _create_folder(syn, "CreateFolderTest", "syn23529995")
>>> out
Folder(createdOn='2021-07-02T18:49:31.396Z', etag='e19c77cd-7813-4f47-9049-4273641a29ed', createdBy='3413689', modifiedOn='2021-07-02T18:49:31.545Z', modifiedBy='3413689', id='syn25946242', concreteType='org.sagebionetworks.repo.model.Folder', parentId='syn23529995', name='CreateFolderTest')
>>> out = _create_folder(syn, "CreateFolderTest", "syn23529995")
>>> out
Folder(createdOn='2021-07-02T18:49:31.396Z', createdBy='3413689', modifiedBy='3413689', id='syn25946242', modifiedOn='2021-07-02T18:49:57.932Z', etag='ea49520d-bf9f-4bb8-a471-803b30330d5a', concreteType='org.sagebionetworks.repo.model.Folder', parentId='syn23529995', name='CreateFolderTest')
>>> out = _create_folder(syn, "CreateFolderTest", "syn23529995")
>>> out
Folder(createdOn='2021-07-02T18:49:31.396Z', modifiedOn='2021-07-02T18:50:26.870Z', createdBy='3413689', modifiedBy='3413689', id='syn25946242', concreteType='org.sagebionetworks.repo.model.Folder', parentId='syn23529995', etag='d33f6432-cda4-443b-88b1-9874f464f96d', name='CreateFolderTest')

>>> syn.restGET("/entity/syn25946242/version")
{'totalNumberOfResults': 1, 'results': [{'id': 'syn25946242', 'versionNumber': 1, 'versionLabel': '1', 'isLatestVersion': True, 'modifiedBy': '3413689', 'modifiedByPrincipalId': '3413689', 'modifiedOn': '2021-07-02T18:50:26.870Z'}]}
```